src/inputs/internal/ComboBoxBase.tsx

```diff
@@ -285,7 +288,12 @@ export function ComboBoxBase<O, V extends Value>(props: ComboBoxBaseProps<O, V>)
         inputValue: getInputValue(selectedOptions, getOptionLabel, multiselect, nothingSelectedText, isReadOnly),
       }));
     }
-  }, [state.isOpen, selectedOptions, getOptionLabel, multiselect, nothingSelectedText, isReadOnly]);
+  }, [state.isOpen, selectedOptions, getOptionLabel, multiselect, nothingSelectedText, isReadOnly, onSearch]);
+
+  // Call on search callback when the user types in the input field
+  useEffect(() => {
+    onSearch?.(fieldState.searchValue ?? "");
+  }, [onSearch, fieldState.searchValue]);
```

Comment:
Question: We should debounce this on the beam side of the API? cc @bdow wdyt?

@0ces just curious, what use case is this for?

Reply:
I did debounce it but on the BP side, I'll change that